### PR TITLE
[X86] Assign AVX10_1 feature priority to align with gcc.

### PR DESCRIPTION
**Pull request by FreddyLeaf**
gcc patch: https://gcc.gnu.org/git/?p=gcc.git;a=commit;h=1f2ca510065a2033bac408eb5a960ef0126f25cc

### CODE REVIEW

File: llvm/include/llvm/TargetParser/X86TargetParser.def [hunk 1]
```diff
@@ -250,8 +250,8 @@ X86_FEATURE_COMPAT(SHA512,          "sha512",                 0)
 X86_FEATURE_COMPAT(SM4,             "sm4",                    0)
 X86_FEATURE       (EGPR,            "egpr")
 X86_FEATURE_COMPAT(USERMSR,         "usermsr",                0)
-X86_FEATURE_COMPAT(AVX10_1,         "avx10.1-256",            0)
-X86_FEATURE_COMPAT(AVX10_1_512,     "avx10.1-512",            0)
+X86_FEATURE_COMPAT(AVX10_1,         "avx10.1-256",           36)
```
Comment (phoebewang) [inline llvm/include/llvm/TargetParser/X86TargetParser.def:253]: Should move them to line 176?

**Review by phoebewang — COMMENTED** [review]
> gcc patch: https://gcc.gnu.org/git/?p=gcc.git;a=commit;h=1f2ca510065a2033bac408eb5a960ef0126f25cc

I don't understand the relationship with GCC, E.g., the GCC patch showes AVX10.1-256/512 are not successors to AVX512VP2INTERSECT but X86_64_V4.

File: llvm/include/llvm/TargetParser/X86TargetParser.def (hunk 1 above)
Comment (FreddyLeaf) [reply llvm/include/llvm/TargetParser/X86TargetParser.def:253]: We can't do so, the location of X86_FEATURE_COMPAT also  has influence. need to align with gcc/compiler-rt

**Comment by FreddyLeaf** [comment]
> > gcc patch: https://gcc.gnu.org/git/?p=gcc.git;a=commit;h=1f2ca510065a2033bac408eb5a960ef0126f25cc
> 
> I don't understand the relationship with GCC, E.g., the GCC patch showes AVX10.1-256/512 are not successors to AVX512VP2INTERSECT but X86_64_V4.

The priorities of AVX512VP2INTERSECT and many others are added by a long time ago in llvm only. I kept them to not breack backward compatibility in llvm.

**Comment by FreddyLeaf** [comment]
ping

**Review by phoebewang — APPROVED** [review]
LGTM.

**Comment by tbaederr** [comment]
I think this broke several codegen/attr tests with assertions enabled:

```
FAIL: Clang :: CodeGen/unique-internal-linkage-names.cpp (11739 of 22485)
******************** TEST 'Clang :: CodeGen/unique-internal-linkage-names.cpp' FAILED ********************
Exit Code: 2

Command Output (stderr):
--
RUN: at line 3: /root/llvm-project/build/bin/clang -cc1 -internal-isystem /root/llvm-project/build/lib/clang/19/include -nostdsysteminc -triple x86_64-linux-gnu -x c++ -emit-llvm -o - < /root/llvm-project/clang/test/CodeGen/unique-internal-linkage-names.cpp | /root/llvm-project/build/bin/FileCheck /root/llvm-project/clang/test/CodeGen/unique-internal-linkage-names.cpp --check-prefix=PLAIN
+ /root/llvm-project/build/bin/clang -cc1 -internal-isystem /root/llvm-project/build/lib/clang/19/include -nostdsysteminc -triple x86_64-linux-gnu -x c++ -emit-llvm -o -
+ /root/llvm-project/build/bin/FileCheck /root/llvm-project/clang/test/CodeGen/unique-internal-linkage-names.cpp --check-prefix=PLAIN
clang: /root/llvm-project/llvm/lib/TargetParser/X86TargetParser.cpp:763: unsigned int llvm::X86::getFeaturePriority(ProcessorFeatures): Assertion `std::is_permutation(HelperList.begin(), HelperList.end(), std::begin(Priorities), std::end(Priorities)) && "Priorities don't form consecutive range!"' failed.
```

**Comment by FreddyLeaf** [comment]
Thanks for reminding, I think I forgot to build check-all debug build again. Sorry for that. Will revert first.

**Pull request merged**